clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -945,24 +945,29 @@ static void addBackendOptions(const ArgList &Args,
   if (IsCPU) {
     OptC.split(CmdArgs, " ", /*MaxSplit=*/-1, /*KeepEmpty=*/false);
   } else {
-    // ocloc -options args need to be comma separated, e.g. `-options
-    // "-g,-cl-opt-disable"`. Otherwise, only the first arg is processed by
-    // ocloc as an arg for -options, and the rest are processed as standalone
-    // flags, possibly leading to errors.
+    // ocloc -options takes arguments in the form of '-options "-g
+    // -cl-opt-disable"' where each argument is separated with spaces.
     // split function here returns a pair with everything before the separator
     // ("-options") in the first member of the pair, and everything after the
     // separator in the second part of the pair. The separator is not included
     // in any of them.
     auto [BeforeOptions, AfterOptions] = OptC.split("-options ");
     // Only add if not empty, an empty arg can lead to ocloc errors.
-    if (!BeforeOptions.empty())
-      CmdArgs.push_back(BeforeOptions);
+    if (!BeforeOptions.empty()) {
+      SmallVector<StringRef, 8> BeforeArgs;
```

Comment:
also just wondering, i see one of your commits is titled says 'revert nick's PR' but i don't remember writing that code so if it is referring to me do you mind linking the PR being reverted, thanks

Reply:
I believe, resolving all SYCL E2E test failures with new offloading model is necessary to enable new model by default, but it is not the only condition.
I think as soon as SYCL E2E pass, we want to enable regular testing (pre-commit? post-commit? nightly?) with new offloading model to make sure no regressions.
Or we might want to even enable it now and mark not-passing tests as XFAIL.
Thoughts?